ts/sagas/startup.ts

```diff
@@ -53,6 +54,9 @@ function* initializeApplicationSaga(): IterableIterator<Effect> {
   // user profile.
   yield put(resetProfileState);
 
+  // Watch for bail outs during the onboarding.
+  yield fork(watchResetOnboardingSaga);
```

Comment:
I would restrict the lifetime of this saga to the onboarding process, forking it from here means it lives forever.

You should probably use [race](https://redux-saga.js.org/docs/advanced/RacingEffects.html) to run this along the onboarding process and stop it once we're out the onboarding process over